bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -400,11 +409,17 @@ public void thingAdded(Thing thing, ThingTrackerEvent thingTrackerEvent) {
         this.things.add(thing);
         logger.debug("Thing '{}' is tracked by ThingManager.", thing.getUID());
 
+        if (storage != null && storage.containsKey(thing.getUID().getAsString())
+                && !storage.get(thing.getUID().getAsString())) {
+            setEnabled(thing.getUID(), false);
+        }
+
```

Comment:
The `setEnabled(...)`-method calls `registerAndInitializeHandler(...)` for enabled things. When looking at this from this place in the code:
* it is pretty confusing when looking at the method name "setEnabled", don't you think so?
* Isn't the below code useless now?